core/common/src/main/scala/org/typelevel/otel4s/KindTransformer.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+import cats.Applicative
+import cats.Functor
+import cats.Monad
+import cats.data.EitherT
+import cats.data.IorT
+import cats.data.Kleisli
+import cats.data.Nested
+import cats.data.OptionT
+import cats.data.StateT
+import cats.effect.MonadCancelThrow
+import cats.effect.Resource
+import cats.implicits.toFunctorOps
+import cats.syntax.all._
+import cats.~>
+
+/** A utility for transforming the higher-kinded type `F` to another
+  * higher-kinded type `G`.
+  */
+@annotation.implicitNotFound("No transformer defined from ${F} to ${G}")
+trait KindTransformer[F[_], G[_]] {
```

Comment:
Which laws `KindTransformer` should obey? Perhaps we can use Discipline to run law tests? https://typelevel.org/cats/typeclasses/lawtesting.html

Reply:
I don't know and that sounds like a great idea!